syntax = "proto3";

package shared_proto;

// Market data tick information.
message MarketTick {
    string symbol     = 1;
    int64  ts_unix_ns = 2; // nanoseconds since epoch
    double bid_price  = 3;
    double ask_price  = 4;
    double last_price = 5;
    int32  bid_size   = 6;
    int32  ask_size   = 7;
    int32  last_size  = 8;
    string venue      = 9; // e.g. "SMART"
}
